go/tools/testwrapper/wrap.go

```diff
@@ -0,0 +1,31 @@
+package testwrapper
+
+import (
+	"bytes"
+	"io"
+	"os"
+
+	"os/exec"
+)
+
+// WrapIfNecessary wraps the execution of the test
```

Comment:
Might be better to split this into two functions so that `main` has more control over how the process exits.

```
func shouldWrap() bool
func wrap() error
```

The call would be like:

```
if shouldWrap() {
  err := wrap()
  if xerr, ok := err.(*exec.ExitError); ok {
    os.Exit(xerr.ExitCode())
  } else if err != nil {
    log.Print(err)
    os.Exit(6) // something distinctive for abnormal termination
  } else {
    os.Exit(0)
  }
}
```